File: postgraduate/src/main/java/com/postgraduate/service/TrainService.java
package com.postgraduate.service;


import java.util.List;


import com.postgraduate.pojo.TbTrain;
import com.postgraduate.pojo.TbLesTra;
import entity.PageResult;

public interface TrainService {
	/**
	 * 新增加培养方案
	 * @return
	 */
	public void insert(TbTrain tbTrain);
	/**
	 * 新增加培养方案课程
	 * @return
	 */
	public void insertLesTra(int lesid,int traid);
	/**
	 * 查询所有培养方案
	 * @return
	 */
	public List<TbTrain> selectAll();
	/**
	 * 查询单个培养方案
	 * @return
	 */
	public TbTrain selectOne(int id);

	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	/**
	 * 模糊查询返回分页列表
	 * @return
	 */
	public PageResult findPage(TbTrain tbTrain,int pageNum,int pageSize);
	/**
	 * 培养方案审核
	 * @return
	 */
	public void updateState(TbTrain tbTrain);
}

File: postgraduate/src/main/webapp/js/service/TrainService.js
app.service('TrainService',function($http){
	//增加培养计划
	this.insert=function(entity){
		return $http.post('../Train/insert.do',entity);
	}
	//单条查询
	this.selectOne=function(id){
		return $http.get('../Train/selectOne.do?id='+id);
	}
	this.selectAll=function(){
		return $http.get('../Train/selectAll.do');
	}
	//分页
	this.findPage=function(page,rows){
		return $http.get('../Train/findPage.do?page='+page+'&rows='+rows);
	}
	//审核
	this.updateState=function(id,state){
		return $http.post('../Train/updateState.do?id='+id+'&state='+state);
	}
	//模糊查询分页
	this.search=function(page,rows,searchEntity){
		return $http.post('../Train/search.do?page='+page+'&rows='+rows,searchEntity);
	}
	this.insertLesTra=function(lesid,traid){
		return $http.post('../Train/insertLesTra.do?lesid='+lesid+'&traid='+traid);
	}
});